src/basilisp/test.lpy

```diff
@@ -73,7 +73,7 @@
   "Implementation detail of :lpy:fn:`is` for generating macros."
   (fn [expr _ _]
     (cond
-      (list? expr) (first expr)
+      (and (sequential? expr) (not (vector? expr))) (symbol (name (first expr)))
```

Comment:
This line made the assumption that `(first expr)` is always a symbol, which seemed dangerous to me. While investigating, I ended up fixing it and incorporating the fix into #831 (which also includes `thrown-with-msg?` assertion).

Reply:
Thanks for looking into it in detail!